### PR TITLE
feat: Allow overriding download behavior using `GCLOUD_TF_DOWNLOAD` environment variable.

### PR DESCRIPTION
**Pull request by morgante**
This should make customizing the behavior easier without plumbing through Terraform modules everywhere.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/terraform-google-gcloud/README.md /tmp/tmp.JjHZZEesmV/generate_docs/workspace/terraform-google-gcloud/README.md
70a71
> | downloaded | Whether gcloud was downloaded or not |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/terraform-google-gcloud/examples/simple_example/README.md /tmp/tmp.JjHZZEesmV/generate_docs/workspace/terraform-google-gcloud/examples/simple_example/README.md
11a12,17
> ## Outputs
> 
> | Name | Description |
> |------|-------------|
> | downloaded | Whether gcloud was downloaded or not |
> 
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
terraform-google-gcloud/outputs.tf
--- old/terraform-google-gcloud/outputs.tf
+++ new/terraform-google-gcloud/outputs.tf
@@ -36,6 +36,6 @@
 
 output "downloaded" {
   description = "Whether gcloud was downloaded or not"
-  value       = !local.skip_download
-  depends_on = [local.wait]
+  value       = ! local.skip_download
+  depends_on  = [local.wait]
 }
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

File: main.tf [hunk 1]
```diff
@@ -16,19 +16,22 @@
...
   tmp_credentials_path = "${path.module}/terraform-google-credentials.json"
-  cache_path           = "${path.module}/cache/${random_id.cache.hex}"
+  cache_path           = local.skip_download ? "" : "${path.module}/cache/${random_id.cache[0].hex}"
   gcloud_tar_path      = "${local.cache_path}/google-cloud-sdk.tar.gz"
   gcloud_bin_path      = "${local.cache_path}/google-cloud-sdk/bin"
   gcloud_bin_abs_path  = abspath(local.gcloud_bin_path)
   components           = join(",", var.additional_components)
 
-  gcloud              = var.skip_download ? "gcloud" : "${local.gcloud_bin_path}/gcloud"
+  download_override = data.external.env_override.result.download
```
Comment (bharathkkb) [inline main.tf:25]: As the `data.external.env_override.result.download` is just used for reading if GCLOUD_TF_DOWNLOAD is set does it make sense to just make it a TF variable? This way we can natively set it with `TF_VAR_*`.

Comment (morgante) [reply main.tf:25]: TF variables are only applied at the root module/workspace. We want to be able to use a global variable without having to pass it through modules.

Comment (bharathkkb) [reply main.tf:25]: Ah yes that escaped me :)

**Pull request merged**